pandas/core/indexes/base.py

```diff
@@ -4020,6 +4020,11 @@ def _add_numeric_methods_binary(cls):
 
         def _make_evaluate_binop(op, opstr, reversed=False, constructor=Index):
             def _evaluate_numeric_binop(self, other):
+                if (isinstance(other, ABCSeries) and
```

Comment:
I don't like this at all. either let's remove this for now (and the corresponding test) or simplemente this by an override in the appopriate subclass.

Reply:
Removing this for now.

Overriding in subclass is a PITA b/c TimedeltaIndex._add_numeric_methods() is called below the definition of TimedeltaIndex and will override any `__mul__` or `__div__` methods we put in the class itself.  So to override we need to implement it outside the class and pin it on after the call to add_numeric_methods.  Since this is among my least favorite patterns, I'm punting on it.